### PR TITLE
Update for embedded URLs

### PR DESCRIPTION
**Pull request by jdf400-rapid7**
### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by jdf400-rapid7** [comment]
## Assessment
### Run

<details>

```
{
  "body": {
    "log": "rapid7/Microsoft Office 365 ATP Safe Links:1.0.3. Step name: decode\n",
    "meta": {},
    "output": {
      "result": "http://www.if2732gl.iwcnllc.com/#https%!A(MISSING)//lojavapor.com.br/ug/?\u0026amp;Pepper=Them\u0026amp;gang%!D(MISSING)Anita.Christian@coreandmain.com\u2029"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/microsoft_atp_safe_links:1.0.3 run < tests/decode_nonsafelink.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

#### Workflow Builder


#### Job


#### Artifact

File: microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def run(self, params={}):
         url = params.get(Input.URL)
         try:
             results = decode_url(url)
-            return {Output.RESULT: results}
+            if results:
+                return {Output.RESULT: results}
+            return {Output.RESULT: url}
```
Comment (mrinehart-r7) [inline microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: I'm not sure returning the original URL is the right solution - how will they know if the decode succeeded?

Comment (jdf400-rapid7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: so if the results are an empty string right now we will just return the empty string and say it decoded when it didnt

Comment (jschipp-r7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: We might want to add a variable called `decoded` that's a boolean to indicate to the user.

Comment (jdf400-rapid7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: wouldnt the user then have to do a {{#if [decoded]}} to use the output of this plugin or the original url?

Comment (jschipp-r7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: If we had the boolean, then they could do an automated decision for `{{decoded}} = true` which is cleaner IMO

Comment (jdf400-rapid7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: reviving this, my thought is that most customers just want to use this in a loop to send to virus total or something. If they have to add an automated decision and have two seperate VT steps its going to make it even harder than it already is to extract the loop data.

Comment (jschipp-r7) [reply microsoft_atp_safe_links/komand_microsoft_atp_safe_links/actions/decode/action.py]: Thanks for bringing this back up.

**Pull request closed without merging**